addons/point_of_sale/static/src/app/models/related_models.js

```diff
@@ -474,652 +463,666 @@ export function createRelatedModels(modelDefs, modelClasses = {}, opts = {}) {
             };
         }
         return new Model({
-            models,
-            records,
-            model: models[model],
-            traps: { set: setTrapsCache[model] },
+            model,
+            traps: { set: setTrapsCache[modelName] },
         });
     }
 
-    const create = withoutProxyTrap(_create);
-    function _create(
-        crud,
-        model,
-        vals,
-        ignoreRelations = false,
-        fromSerialized = false,
-        delayedSetup = false
-    ) {
-        if (!("id" in vals)) {
-            vals["id"] = uuid(model);
+    /**
+     * A model (e.g. pos.order) points to an instance of this class.
+     */
+    class CRUD {
+        constructor(name) {
+            this.name = name;
         }
+        get models() {
+            return models;
+        }
+        get records() {
+            return records;
+        }
+        get fields() {
+            return getFields(this.name);
+        }
+        get orderedRecords() {
+            return Array.from(this.records[this.name].values());
+        }
+        instantiate() {
+            return instantiate(this);
+        }
+        exists(id) {
+            return this.records[this.name].has(id);
+        }
+        create(vals, ignoreRelations = false, fromSerialized = false, delayedSetup = false) {
+            const record = disabler.call(
+                (...args) => this._create(...args),
+                vals,
+                ignoreRelations,
+                fromSerialized,
+                delayedSetup
+            );
+            this.models.makeRecordsAvailable({ [this.name]: [record] }, { [this.name]: [vals] });
+            return record;
+        }
+        deserialize(vals) {
+            return this.create(vals, false, true);
+        }
+        createMany(valsList) {
+            const result = [];
+            for (const vals of valsList) {
+                result.push(this.create(vals));
+            }
+            return result;
+        }
+        update(record, vals) {
+            return disabler.call((...args) => this._update(...args), record, vals);
+        }
+        delete(record, opts = {}) {
+            return disabler.call((...args) => this._delete(...args), record, opts);
+        }
+        deleteMany(toDelete) {
+            const result = [];
+            let mustBreak = 0;
+            while (toDelete.length) {
+                result.push(this.delete(toDelete[toDelete.length - 1]));
+                mustBreak += 1;
+
+                if (mustBreak > 1000) {
+                    console.warn("Too many records to delete. Breaking the loop.");
+                    break;
+                }
+            }
+            return result;
+        }
+        read(value) {
+            const id = /^\d+$/.test(value) ? parseInt(value) : value; // In case of ID came from an input
```

Comment:
I think is more performant to do just a 

```js
typeof value === "number"
```

Reply:
can't do that. Seems like the `read` allows both the integer `11` and the string `"11"`.